scanpy/plotting/_tools/scatterplots.py

```diff
@@ -756,9 +756,16 @@ def spatial(
     -------
     If `show==False` a :class:`~matplotlib.axes.Axes` or a list of it.
     """
+    library_id = list(adata.uns['spatial'].keys())[0]
+
     if img_key is _empty:
         img_key = next(
-            (k for k in ['hires', 'lowres'] if k in adata.uns['images']), None
+            (
+                k
+                for k in ['hires', 'lowres']
+                if k in adata.uns['spatial'][library_id]['images']
+            ),
+            None,
```

Comment:
What's the intended behavior here? There's a priority for `hires`, `lowres`, and then what happens?

Also, what's the case where `"hires"` doesn't exist?

Reply:
Yeah, priority completely arbitrary, this behaviour kicks in only if there is no `img_key` specified, so that at least it tried to plot an image below. Otherwise it's just None and it does not plot any image, only the points in spatial coordinates.

Also, these additions were done only to make the tests pass.

Reply:
What test was failing?

Reply:
test_embedding.py